### PR TITLE
Update DeleteOperation to handle at least one replica success properly

### PR DESCRIPTION
**Pull request by mikeyang923**
Return 503 instead of 404 when one replica returns success but others return not_found for DELETE operation

### CODE REVIEW

File: ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java [hunk 1]
```diff
@@ -361,8 +361,15 @@ private void checkAndMaybeComplete() {
             new RouterException("DeleteOperation failed possibly because some replicas are unavailable",
                 RouterErrorCode.AmbryUnavailable));
       } else if (operationTracker.hasFailedOnNotFound()) {
-        operationException.set(
-            new RouterException("DeleteOperation failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist));
+        if (operationTracker.getSuccessCount() > 0) {
```
Comment (justinlin-linkedin) [inline ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java:364]: this is all good, but can you add some test cases just like the ttlupdate?

Comment (mikeyang923) [reply ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java:364]: Done in commit [796d906](https://github.com/linkedin/ambry/pull/2208/commits/796d906360d7bd66620014173d65915cdff39962)

[the remark this answers is not in the document]
Comment (mikeyang923) [reply ambry-router/src/main/java/com/github/ambry/router/DeleteOperation.java:370]: Added comments in the code

**Review by ankagrawal — APPROVED** [review]
Approving this PR. It covers majority of the cases where we have seen 404s being returned for Delete and TtlUpdate operations.

**Review by justinlin-linkedin — APPROVED** [review]

**Pull request merged**